src/libs/ajax/DatasetBuilder.ts

```diff
@@ -148,13 +174,51 @@ const getDummyConcepts = async (parent: Concept): Promise<GetConceptsResponse> =
   };
 };
 
-export const DatasetBuilder = (): DatasetBuilderContract => ({
-  retrieveDataset: async (datasetId) => {
-    return await Ajax()
-      .DataRepo.dataset(datasetId)
-      .details([datasetIncludeTypes.SNAPSHOT_BUILDER_SETTINGS, datasetIncludeTypes.PROPERTIES]);
-  },
-  getConcepts: (parent: Concept) => Promise.resolve(getDummyConcepts(parent)),
-  requestAccess: (_request) => Promise.resolve(),
-  getParticipantCount: (_request) => Promise.resolve(100),
-});
+export const DatasetBuilder = (): DatasetBuilderContract => {
+  return {
+    retrieveDataset: async (datasetId) => {
+      return await Ajax()
+        .DataRepo.dataset(datasetId)
+        .details([datasetIncludeTypes.SNAPSHOT_BUILDER_SETTINGS, datasetIncludeTypes.PROPERTIES]);
+    },
+    getProgramDataStatistics: async (datasetId, programDataOption) => {
+      switch (programDataOption.kind) {
```

Comment:
I wasn't sure if this logic should belong here or higher up. I think it will eventually be more dependent on specific APIs called for different types of data, but its definitely still UI business logic.

Reply:
I think it's OK to keep it here but I think it would read better if we could keep the object conversion code out of the API calls, by refactoring if possible.

Reply:
I second that the contract would be more readable if refactored